src/pylint_odoo/checkers/odoo_addons.py

```diff
@@ -1009,6 +1015,12 @@ def visit_call(self, node):
                 # Check just the following cases "%s %s..."
                 self.add_message("translation-positional-used", node=node, args=(str2translate,))
 
+            # translation-replace: recommend to translate a string (_) with self.env._
+            if isinstance(arg, nodes.Const) and not isinstance(
+                node.func, nodes.Attribute
+            ):  # ensure it's not called as attribute, e.g: self.env._()
+                self.add_message("prefer-env-translation", node=node)
```

Comment:
to make sure that it can really be replaced by `self.env._`, can you also had a check that goes through `node.parent` to make sure there is a `self` or a `env`?
because there are some cases where `_()` is called without any env around

Reply:
Hi anh, true, I added some checks to make sure it impacts correct places. Do you think I miss anything?